pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ServerCnxTest.java

```diff
@@ -1030,6 +1030,55 @@ public void testVerifyAuthRoleAndAuthDataFromDirectConnectionBroker() throws Exc
                 }));
     }
 
+    @Test
+    public void testRefreshOriginalPrincipalWithAuthDataForwardedFromProxy() throws Exception {
```

Comment:
Note that the tests that comment about `https://github.com/apache/pulsar/issues/19332` should fail because they make assertions on the current behavior.

Reply:
For current design, it works fine.

Reply:
I must not have written some of the assertions I thought I did. You're right that those tests all pass. It might be worth removing the comments that reference #19332 because your PR will make them incorrect.